ci/travis/format.sh

```diff
@@ -26,23 +26,36 @@ FLAKE8_VERSION=$(flake8 --version | awk '{print $1}')
 YAPF_VERSION=$(yapf --version | awk '{print $2}')
 
 if [[ $FLAKE8_VERSION != "3.7.7" ]]; then
-  echo "Please install flake8==3.7.7. You currently are using $FLAKE8_VERSION."
-  exit 1
+    echo "WARNING: Ray uses flake8==3.7.7. You currently are using $FLAKE8_VERSION." \
+         "This might generate different results."
+    read -p "Do you want to continue?[y/n]" input
+    if ! [ $input = 'y' ]; then
+        exit 1
+    fi
 fi
 
 if [[ $YAPF_VERSION != "0.23.0" ]]; then
-  echo "Please install yapf==0.23.0. You currently are using $YAPF_VERSION."
-  exit 1
+    echo "WARNING: Ray uses yapf==0.23.0. You currently are using $YAPF_VERSION." \
+         "This might generate different results."
+    read -p "Do you want to continue?[y/n]" input
+    if ! [ $input = 'y' ]; then
+        exit 1
+    fi
 fi
 
 if which clang-format >/dev/null; then
   CLANG_FORMAT_VERSION=$(clang-format --version | awk '{print $3}')
   if [[ $CLANG_FORMAT_VERSION != "7.0.0" ]]; then
-    echo "Please install clang-format 7.0.0. You currently are using $CLANG_FORMAT_VERSION."
-    exit 1
+    echo "WARNING: Ray uses clang-format 7.0.0, but you are using $CLANG_FORMAT_VERSION." \
+         "This might generate different results."
+    read -p "Do you want to continue? [y/n]" input
+    if ! [ $input = 'y' ]; then
+        exit 1
+    fi
   fi
 else
-  echo "WARNING: clang-format is not installed!"
+    echo "WARNING: clang-format is not installed!"
+    exit 1
```

Comment:
I didn't exit here before because I thought some Python developers (e.g., for RLlib) might not install `clang-format`. It might be better to warn them and then ask if they want to proceed?